src/gateway_config_miner.erl

```diff
@@ -0,0 +1,98 @@
+-module(gateway_config_miner).
+
+-behavior(gen_server).
+
+-include("gateway_config.hrl").
+-define(SERVICE, 'helium.local.api').
+
+%% gen_server
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2
+]).
+
+%% api
+-export([
+    status/0,
+    pubkey/0,
+    add_gateway/3
+]).
+
+-record(state, {
+    connection = undefined :: grpc_client:connection() | undefined
+}).
+
+-spec status() -> [tuple()].
+status() ->
+    gen_server:call(?MODULE, status).
+
+-spec pubkey() -> {PubKey :: binary(), OnboardingKey :: binary()}.
+pubkey() ->
+    gen_server:call(?MODULE, pubkey).
+
+-spec add_gateway(
+    Owner :: libp2p_crypto:pubkey_bin(),
+    Payer :: libp2p_crypto:pubkey_bin(),
+    Mode :: full | dataonly | light
+) -> {ok, binary()} | {error, term()}.
+add_gateway(Owner, Payer, Mode) ->
+    gen_server:call(?MODULE, {add_gateway, [Owner, Payer, Mode]}).
+
+init(_) ->
+    erlang:register(?MODULE, self()),
+    {ok, #state{}}.
+
+handle_call(Call, From, State = #state{connection = undefined}) ->
+    case grpc_client:connect(tcp, "localhost", 4467) of
+        {ok, Connection} ->
+            handle_call(Call, From, State#state{connection = Connection});
```

Comment:
in principle i like only establishing the connection when it's actually needed, but i think you still want to throw a monitor on the http_connection pid that comes back as part of that connection (`{ok, #{http_connection := ConnPid} = Connection} = grpc_client:connect(tcp, "localhost", 4467)`) and gc the connection ~~if~~ when it goes down; we've seen that grpc connection pid die but the app will continue to use a connection record with a dead http connection which just continually fails and since you're handling the error response i don't think it'll crash and restart the genserver to start over.

in order to reduce the long-standing and little-used connections on the local grpc api would it also be helpful to throw a `send_after` in here as well and disconnect the grpc connection so it's less likely to crash randomly and cause any hiccups in between these infrequent calls?